### PR TITLE
DTO implementation and BlockGenerator profile added

### PR DESCRIPTION
**Pull request by TheGreyCore**

### CODE REVIEW

File: src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java [hunk 1]
```diff
@@ -38,21 +39,20 @@ public TransactionsController(TransactionService transactionService) {
...
+     * @param newTransactionDTO This is the transaction to be created.
      * @return returns the result of the transaction creation.
      */
+    @Valid
     @PostMapping("/send")
-    public ResponseEntity<?> newTransactions(@Valid @RequestBody Transaction transaction) {
-        // Custom constraint validation using @Valid annotation on the arguments works as expected.
-        // All custom ConstraintValidators have their dependencies injected as expected.
-        return ResponseEntity.ok().body(transactionService.newTransactions(transaction));
+    public ResponseEntity<Integer> newTransactions(@RequestParam NewTransactionDTO newTransactionDTO){
```
Comment (inugami-dev64) [inline src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java]: Use @Valid annotation here to validate incoming transaction objects before having potentially invalid transactions reach the service layer.

File: src/main/java/org/students/simplebitcoinnode/service/TransactionService.java [hunk 2]
```diff
@@ -45,15 +56,12 @@ public List<Transaction> getTransactions(String pubKey, String type) {
      *             It returns 201 if the transaction was successfully created, and 400 if an exception occurred.
      */
     
-    public int newTransactions(Transaction transaction) {
+    public int newTransactions(NewTransactionDTO newTransactionDTO) {
```
Comment (inugami-dev64) [inline src/main/java/org/students/simplebitcoinnode/service/TransactionService.java]: Method `newTransactions` shouldn't return an integer describing the HTTP status code but rather the `GetTransactionDTO` object describing the transaction that was saved to the database. Service layer shouldn't know anything about the underlying protocol details used by the controller.

File: src/main/resources/exampleRequests/getExample.http [hunk 3]
```diff
@@ -0,0 +1 @@
+GET http://localhost:8080/blockchain/transactions?pubKey=PZ8Tyr4Nx8MHsRAGMpZmZ6TWY63dXWSCwaXBRYUfimKYY74i3QDYj6Y2rFgb6tGHmMNUGZssGx71FB1M2KnDMugTE2h6N8ttkpxSrT7arsC4Lqum3zrkF3oJ&type=sent
```
Comment (inugami-dev64) [inline src/main/resources/exampleRequests/getExample.http]: Remove this file, since the HTTP GET query depends on the test key, which is randomly generated and only exists on your machine.

File: src/main/java/org/students/simplebitcoinnode/service/DTOMapperService.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class DTOMapperService {
+    private final ModelMapper mapper = new ModelMapper();
```
Comment (inugami-dev64) [inline src/main/java/org/students/simplebitcoinnode/service/DTOMapperService.java]: I would not call this class a "service" but rather a "wrapper", since we are essentially using decorator pattern to extend the functionality of ModelMapper. Rename this class and move it to some other subpackage e.g `org.students.util.wrapper` or smth similar. Also please create unit tests for this class.

**Review by inugami-dev64 — CHANGES_REQUESTED** [review]
Do not use camelCase notation for naming Java packages, use all lowercase naming instead and call it `dto` for example.
https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html
The code has some issues that need to be resolved before I can merge it to master. Read diff comments for more information.

File: src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java (hunk 1 above)
Comment (TheGreyCore) [reply src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java]: Already there. 
image

Comment (inugami-dev64) [reply src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java]: That's not the correct usage of `@Valid` annotation because it does not validate the request parameter but instead validates the controller return object. Apply `@Valid` annotation to the method argument instead. Also I just noticed that you are using `@RequestParam` for controller argument, which essentially maps url variables to the object, which is not the desired usage of the REST API. Data about new transactions should be contained in the request body and as such please replace this annotation with `@RequestBody`.

TL;DR do this:
```java
public ResponseEntity<?> newTransactions(@Valid @RequestBody NewTransactionDTO newTransactionDTO) {
```

Comment (TheGreyCore) [reply src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java]: Mhm, fixed typo

**Review by inugami-dev64 — APPROVED** [review]
Well done
